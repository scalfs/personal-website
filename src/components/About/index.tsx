import Title from 'components/Title'
import Image from 'next/image'

const About = () => (
  <section className="pt-8 pb-28" id="about">
    <div className="md:flex md:flex-row-reverse md:gap-12">
      <div className="mb-8 space-y-4 text-justify md:mb-0 md:w-3/5">
        <Title>About Me</Title>
        <p>
          My first contact with programming was as a teenager, but I got to
          really explore it in Electrical Engineering college. During my time
          there I was part of a research lab working with robotics allied to
          computer vision and did my final project in speech processing using
          recurrent neural networks.
        </p>
        <p>
          Started my professional journey as an intern at a large e-commerce
          company, being hired shortly after. Had a great experience working
          with app development and felt like I&apos;d found my calling. Later, I
          had a terrific opportunity to lead the mobile project on an inventory
          management tech company with customers from all over the world.
        </p>
        <p>
          I&apos;m constantly learning new things, specializing in frontend
          while broadening my knowledge in other areas. When I&apos;m not
          working or in front of the computer, you may find me running, riding
          my bike everywhere, and spending time outdoors in nature. Otherwise,
          I&apos;m mostly watching movies/series with my wife, cooking, playing
          the guitar, or entertaining our cat Juju.
        </p>
      </div>

      <div className="m-auto w-3/5 transition-transform duration-250 ease-in-out-cubic hover:-translate-y-1 motion-reduce:transition-none motion-reduce:hover:transform-none md:w-2/5">
        <Image
          width={400}
          height={400}
          src="/img/profile.png"
          alt="Picture of the author. A half smiling professional man."
          className="squircle bg-gradient-to-r from-[#4586e2] to-[#214bc1] object-cover object-center grayscale transition-colors duration-250 ease-in-out-cubic hover:grayscale-0"
        />
      </div>
    </div>

    <svg width="10" height="10" viewBox="0 0 10 10" aria-hidden>
      <clipPath id="squircleClip" clipPathUnits="objectBoundingBox">
        <path
          fill="red"
          stroke="none"
          d="M 0,0.5 C 0,0 0,0 0.5,0 S 1,0 1,0.5 1,1 0.5,1 0,1 0,0.5"
        />
      </clipPath>
    </svg>
  </section>
)

export default About
