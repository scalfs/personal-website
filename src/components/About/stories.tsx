import { ComponentMeta, ComponentStory } from '@storybook/react'
import About from '.'

export default {
  title: 'About',
  component: About
} as ComponentMeta<typeof About>

const Story: ComponentStory<typeof About> = () => <About />

export const Default = Story.bind({})
