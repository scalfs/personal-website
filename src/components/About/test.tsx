import { render, screen } from '@testing-library/react'

import About from '.'

describe('<About />', () => {
  it('should render the section', () => {
    const { container } = render(<About />)

    expect(screen.getByRole('heading', { name: /About/i })).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /Picture/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
