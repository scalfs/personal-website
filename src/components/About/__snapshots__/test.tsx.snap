// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render the section 1`] = `
<section
  class="pt-8 pb-28"
  id="about"
>
  <div
    class="md:flex md:flex-row-reverse md:gap-12"
  >
    <div
      class="mb-8 space-y-4 text-justify md:mb-0 md:w-3/5"
    >
      <h1
        class="mb-4 text-4xl font-extrabold tracking-tight sm:mb-8 sm:text-5xl lg:text-6xl"
      >
        About Me
      </h1>
      <p>
        My first contact with programming was as a teenager, but I got to really explore it in Electrical Engineering college. During my time there I was part of a research lab working with robotics allied to computer vision and did my final project in speech processing using recurrent neural networks.
      </p>
      <p>
        Started my professional journey as an intern at a large e-commerce company, being hired shortly after. Had a great experience working with app development and felt like I'd found my calling. Later, I had a terrific opportunity to lead the mobile project on an inventory management tech company with customers from all over the world.
      </p>
      <p>
        I'm constantly learning new things, specializing in frontend while broadening my knowledge in other areas. When I'm not working or in front of the computer, you may find me running, riding my bike everywhere, and spending time outdoors in nature. Otherwise, I'm mostly watching movies/series with my wife, cooking, playing the guitar, or entertaining our cat Juju.
      </p>
    </div>
    <div
      class="m-auto w-3/5 transition-transform duration-250 ease-in-out-cubic hover:-translate-y-1 motion-reduce:transition-none motion-reduce:hover:transform-none md:w-2/5"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27400%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Picture of the author with a serious yet welcoming expression."
          class="rounded-lg bg-gradient-to-r from-[#4586e2] to-[#214bc1] object-cover object-center grayscale transition duration-250 ease-in-out-cubic hover:grayscale-0"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</section>
`;
